src/spatialdata/dataloader/graph_loader.py

```diff
@@ -0,0 +1,147 @@
+from __future__ import annotations
+
+from typing import Any, Literal
+
+import geopandas as gpd
+import numpy as np
+from scipy.spatial import KDTree
+
+from spatialdata._logging import logger
+from spatialdata._types import ArrayLike
+
+
+def _compute_percentile(tree: KDTree, percentile: int, centroids: ArrayLike) -> float:
+    logger.warning("computing percentile could take long for large datasets.")
+    dd = tree.sparse_distance_matrix(
+        tree,
+        max_distance=np.sqrt(
+            (np.max(centroids[:, 0]) - np.min(centroids[:, 0])) ** 2
+            + (np.max(centroids[:, 1]) - np.min(centroids[:, 1])) ** 2
+        ),
+    ).toarray()
+    np.fill_diagonal(dd, np.nan)
+    dd = dd[~np.isnan(dd)].reshape(dd.shape[0], dd.shape[1] - 1)
+    f = np.percentile(dd, percentile)
+    assert isinstance(f, float)
+    return f
+
+
+def build_graph(  # TODO: add rings for grids
```

Comment:
hi guys, was just taking a look at this and was wondering what was scope, cause it seems like it provides a lot of redundant functionality with squidpy.